File: bookhub_store_front/src/dtos/order/OrderListResponse.dto.ts
export interface OrderListResponseDto {
  orderId: number;
  orderNumber: string;
  orderName: string;

  recipientName: string;
  phoneNumber: string;
  fullAddress: string;
  addressDetail: string;

  totalAmount: number;
  paymentMethod: string;
  orderDate: string;
  status: string;

  orderDetails: OrderDetailResponseDto[];
}

export interface OrderDetailResponseDto {
  orderDetailId: number;
  coverUrl: string;
  bookTitle: string;
  bookPrice: number;
  quantity: number;
  totalPrice: number;
}

File: bookhub_store_back/src/main/java/com/study/bookhub_store_back/dto/order/response/OrderListResponseDto.java
package com.study.bookhub_store_back.dto.order.response;

import com.study.bookhub_store_back.common.enums.OrderStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Builder
@AllArgsConstructor
public class OrderListResponseDto {
    private Long orderId;
    private String orderNumber;
    private String orderName;
    private String customerEmail;
    private String address;
    private Long totalAmount;
    private String paymentMethod;
    private LocalDateTime orderDate;
    private OrderStatus status;

    private List<OrderDetailResponseDto> orderDetails;
}
